### PR TITLE
tests refactor

### PR DESCRIPTION
**Pull request by hiebj**
I also added "black" as a code formatter using a pre-commit hook, and configured the "prettier" code formatter to run using the Python pre-commit tool. Make sure you run `pipenv run pre-commit install` to have the hook installed so that it runs automatically.

The formatter isn't added to README because changes to infrastructure are forthcoming again, and the docs will be updated at that point.

I am going to be bringing in the top-level `infrastructure/` stuff as a git submodule as well as the `start.sh`,`test.unit.sh` and some other things... trying to simplify this project to contain only code specific to this domain.

### CODE REVIEW

File: microservices/events/cmc_calendar/calendar_ingest.py [hunk 1]
```diff
@@ -1,58 +1,37 @@
 #!/usr/bin/env python3.8
-'''
```
Comment (hiebj) [inline microservices/events/cmc_calendar/calendar_ingest.py:2]: Most of this stuff was just done by `black`. If you don't like black or would prefer e.g. `autopep8` we can do that instead, but I hate having to think about formatting in any language, and I like opinionated formatters like black and prettier.

File: microservices/events/cmc_calendar/calendar_ingest.py [hunk 2]
```diff
@@ -1,58 +1,37 @@
...
-    def __init__(self):
-
-        # Configure logger to output to file.
-        logging.basicConfig(filename='calendar-ingest.log')
-
-        # Load the configuration.
-        self.config_parser = ConfigParser()
-        self.config_parser.read('configuration.ini')
-        self.calendarUri = self.config_parser['DEFAULT']['calendarUri']
-        self.eventsBaseUri = self.config_parser['DEFAULT']['eventsBaseUri']
```
Comment (hiebj) [inline microservices/events/cmc_calendar/calendar_ingest.py:30]: Most of this configuration is done in `main.py` now.

The logging configuration (to output to a file) was just removed. Cloud Functions actually like it if you output your log to console - they automatically pick up STDOUT and STDERR and pipe them to Google StackDriver, the log aggregation tool.

File: microservices/events/cmc_calendar/calendar_ingest.py [hunk 3]
```diff
@@ -1,58 +1,37 @@
...
-            cred = credentials.Certificate('cmc-calendar-297516-firebase-adminsdk-mwugq-e33b3b9a7a.json')
-            firebase_admin.initialize_app(cred)
-
-        self.database = firestore.client()
+    """
 
+    def __init__(self, configuration, repository):
+        self.calendarUri = configuration["calendarUri"]
+        self.eventsBaseUri = configuration["eventsBaseUri"]
+        self.repository = repository
```
Comment (hiebj) [inline microservices/events/cmc_calendar/calendar_ingest.py:22]: Now we just accept config, and a "repository" (implemented in `data/firestore_events.py`

**Comment by Transmuted-Space** [comment]
I think you might be overwriting remote refs with force pushes, any particular reason you favor that versus a standard push or --force-with-lease? I had to nuke my local repo because I couldn't get it on track with remote, I think my local refs were totally detached from history.

File: microservices/events/cmc_calendar/calendar_ingest.py (hunk 1 above)
Comment (Transmuted-Space) [reply microservices/events/cmc_calendar/calendar_ingest.py:2]: Agreed, I've not used black but it looks great to me.

**Review by Transmuted-Space — APPROVED** [review]

**Pull request merged**